src/services/crawl-dids/did_processor.service.ts

```diff
@@ -0,0 +1,236 @@
+import { Action, Service } from "@ourparentcenter/moleculer-decorators-extended";
+import { ServiceBroker } from "moleculer";
+import BullableService from "../../base/bullable.service";
+import { DidEventTypes, SERVICE } from "../../common";
+import { calculateDidDeposit } from "../../common/utils/calculate_deposit";
+import { addYearsToDate, formatTimestamp } from "../../common/utils/date_utils";
+
+
+
+interface DidAttribute {
+    key: string;
+    value: string;
+}
+
+interface DidEvent {
+    type: string;
+    did?: string;
+    attributes?: DidAttribute[];
+    [key: string]: any;
+}
+
+interface DidMessageTypes {
+    event_type: string;
+    did?: string;
+    created?: string;
+    modified?: string;
+    deleted_at?: string | null;
+    height?: number;
+    is_deleted?: boolean;
+    exp?: string;
+    deposit?: string;
+    years?: string;
+    id?: string;
+    changes?: any;
+    [key: string]: any;
+}
+
+@Service({
+    name: SERVICE.V1.DidMessageProcessorService.key,
+    version: 1,
+})
+export default class DidMessageProcessorService extends BullableService {
+    constructor(broker: ServiceBroker) {
+        super(broker);
+    }
+
+
+    private computeChanges(
+        oldRec: Record<string, any>,
+        newRec: Record<string, any>
+    ): Record<string, { old: any; new: any }> {
+        const diff: Record<string, { old: any; new: any }> = {};
+        Object.keys(newRec).forEach((key) => {
+            if (newRec[key] !== oldRec[key]) {
+                diff[key] = { old: oldRec[key], new: newRec[key] };
+            }
+        });
+        return diff;
+    }
+
+
+
+    private async saveHistory(didEvent: DidMessageTypes, changes?: any) {
+        const { modified, ...cleanedEvent } = didEvent;
+        await this.broker.call(`${SERVICE.V1.DidHistoryService.path}.save`, {
+            ...cleanedEvent,
+            changes: changes ? JSON.stringify(changes) : null,
+        });
+    }
+
+
+    @Action({ name: "handleDidMessages" })
+    async handleDidMessages(ctx: { params: { listDidTx: DidEvent[] } }) {
+        const { listDidTx } = ctx.params;
+
+        for (const event of listDidTx) {
```

Comment:
Since these are messages, I think the proper naming for `event` and `listDidTx` is simply message and messages respectively. I know this was inherited from previous approach using WS and events, but I think it is important to make sure the naming is consistent to prevent confusion from readers that do not know about the history of this development.

```suggestion
        for (const message of messages) {
```

Reply:
done